src/Proto.Cluster/Gossip/GossipActor.cs

```diff
@@ -24,11 +25,30 @@ public class GossipActor : IActor
         public Task ReceiveAsync(IContext context) => context.Message switch
         {
             SetGossipStateKey setState  => OnSetGossipStateKey(context, setState),
+            GetGossipStateRequest getState => OnGetGossipStateKey(context, getState),
             GossipRequest gossipRequest => OnGossipRequest(context, gossipRequest),
             SendGossipStateRequest      => OnSendGossipState(context),
             _                           => Task.CompletedTask
         };
 
+        private Task OnGetGossipStateKey(IContext context, GetGossipStateRequest getState)
+        {
+            var entries = ImmutableDictionary<string, Any>.Empty;
+            var key = getState.Key;
+            
+            foreach (var (memberId, memberState) in _state.Members)
+            {
+                if (memberState.Values.TryGetValue(key, out var value))
+                {
+                    entries = entries.SetItem(memberId, value.Value);
```

Comment:
for each member in state, try pick the given state key, and add it to the result